src/psyclone/tests/psyir/frontend/fparser2_interface_block_test.py

```diff
@@ -97,11 +97,11 @@ def test_named_interface(fortran_reader, mod_txt):
 
 
 @pytest.mark.parametrize("mod_txt", ["", "module "])
-def test_named_interface_declared(fortran_reader, mod_txt):
+def test_named_interface_declared(fortran_reader, fortran_writer, mod_txt):
     ''' Test that the frontend creates a RoutineSymbol of
     UnsupportedFortranType for a named interface block when the symbol
     name has already been declared (as will be the case for a structure
-    constructor). '''
+    constructor or a subroutine with the same name). '''
```

Comment:
I'm surprised that Fortran lets you do this. I guess it disambiguates by looking at the routine signatures?

Reply:
Yeah, it also surprised me, but this was the case that make it fail, it is in NEMOv5:
```
src/OCE/DYN/divhor.F90
40:   INTERFACE div_hor
41:      MODULE PROCEDURE div_hor, div_hor_old
42:   END INTERFACE
```